src/xhr.js

```diff
@@ -55,7 +67,11 @@ class Xhr {
    * @return {!Promise<!JSONValue>}
    */
   fetchJson(input, opt_init) {
-    return this.fetch_.call(null, input, opt_init).then(response => {
+    if (opt_init && opt_init.method &&
+        opt_init.method.toUpperCase() == 'POST') {
+        setupJsonPost_(opt_init);
```

Comment:
ah ok, if you're ok with that, it would allow me to do assertions.

The reason why I avoided that was i think one of your comments said you wanted `opt_init` optional until getting to `createXhrRequest`